extensions/data-plane-selector/data-plane-selector-api/src/main/java/org/eclipse/edc/connector/dataplane/selector/transformer/JsonObjectToDataPlaneInstanceTransformer.java

```diff
@@ -71,6 +72,10 @@ private void transformProperties(String key, JsonValue jsonValue, DataPlaneInsta
                 var set = jsonValue.asJsonArray().stream().map(jv -> transformString(jv, context)).collect(Collectors.toSet());
                 builder.allowedSourceTypes(set);
             }
+            case ALLOWED_TRANSFER_TYPES -> {
+                var set = jsonValue.asJsonArray().stream().map(jv -> transformString(jv, context)).collect(Collectors.toSet());
```

Comment:
wouldn't it be more robust to use `transformArray` or `transformArrayOrObject` instead? IIRC they should be able to handle string arrays

Reply:
Not sure we handle correctly a `transforArray` or `transformArrayOrObject` with `String.class`